### PR TITLE
Qanary component qanary-component-QB-DateOfDeathDBpedia 

### PR DESCRIPTION
**Pull request by dschiese**
Pull request for new component qanary-component-QB-DateOfDeathDBpedia

### CODE REVIEW

File: qanary-component-QB-DateOfDeathDBpedia/src/test/java/eu/wdaqua/qanary/component/mypackage/QanaryServiceControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+                            String queryTemplate = Files.readString(Paths.get("src/main/resources/queries/insert_one_annotation.rq"));
+                            // probably test with queryTemplate (?)
+                        }
+                    }
+            );
+        }
+
+        //TODO:
+        @Test
+        public void createQueriesWhenNull() {
```
Comment (anbo-de) [inline qanary-component-QB-DateOfDeathDBpedia/src/test/java/eu/wdaqua/qanary/component/mypackage/QanaryServiceControllerTest.java:207]: @dschiese Please remove this test or implement it.

**Review by anbo-de — CHANGES_REQUESTED** [review]
@dschiese The implementation is well-structured and easy to follow. Very good. I added some comments regarding smaller improvements and one refactoring.

File: qanary-component-QB-DateOfDeathDBpedia/src/main/java/eu/wdaqua/qanary/component/dateofdeathrealpersons/qb/QueryBuilderDateOfDeathDBpedia.java [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+        querySolutionMap.add("startValue", ResourceFactory.createTypedLiteral(String.valueOf(supportedPrefix.length()), XSDDatatype.XSDnonNegativeInteger));
+
+        return querySolutionMap;
+    }
+
+
+    public String getDbpediaQuery(String dbpediaResource) throws IOException {
+
+        QuerySolutionMap bindingsForDbpediaQuery = new QuerySolutionMap();
+        bindingsForDbpediaQuery.add("dbpediaResource", ResourceFactory.createResource(dbpediaResource));
```
[the remark this answers is not in the document]
Comment (dschiese) [reply qanary-component-QB-DateOfDeathDBpedia/src/main/java/eu/wdaqua/qanary/component/dateofdeathrealpersons/qb/QueryBuilderDateOfDeathDBpedia.java:147]: image
This leas to this:
image
Fixed it by checking for the correct syntax of an IRI:
image

File: qanary-component-QB-DateOfDeathDBpedia/src/main/java/eu/wdaqua/qanary/component/dateofdeathrealpersons/qb/controller/DefaultController.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import java.io.IOException;
+
+@Controller
+public class DefaultController {
```
Comment (anbo-de) [inline qanary-component-QB-DateOfDeathDBpedia/src/main/java/eu/wdaqua/qanary/component/dateofdeathrealpersons/qb/controller/DefaultController.java:14]: @dschiese Please rename to `QueryBuilderDateOfDeathDBpediaController` to express the connection to `QueryBuilderDateOfDeathDBpedia`.

File: qanary-component-QB-DateOfDeathDBpedia/src/main/java/eu/wdaqua/qanary/component/dateofdeathrealpersons/qb/controller/DefaultController.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+    @Autowired
+    private QueryBuilderDateOfDeathDBpedia queryBuilderDateOfDeathDBpedia;
+
+    @GetMapping("/getdbpediaquery")
```
Comment (anbo-de) [inline qanary-component-QB-DateOfDeathDBpedia/src/main/java/eu/wdaqua/qanary/component/dateofdeathrealpersons/qb/controller/DefaultController.java:19]: @dschiese Please add an OpenAPI annotation here to make it visible on the swagger-ui.html page.

**Comment by anbo-de** [comment]
@dschiese Please see my review comments.

File: qanary-component-QB-DateOfDeathDBpedia/src/main/java/eu/wdaqua/qanary/component/dateofdeathrealpersons/qb/QueryBuilderDateOfDeathDBpedia.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (dschiese) [reply qanary-component-QB-DateOfDeathDBpedia/src/main/java/eu/wdaqua/qanary/component/dateofdeathrealpersons/qb/QueryBuilderDateOfDeathDBpedia.java]: Fixed it here: https://github.com/WDAqua/Qanary-question-answering-components/commit/24b8c5c8f04c3b36af6e5d1cd967c049957e7096

File: qanary-component-QB-DateOfDeathDBpedia/src/main/java/eu/wdaqua/qanary/component/dateofdeathrealpersons/qb/controller/DefaultController.java (hunk 3 above)
Comment (dschiese) [reply qanary-component-QB-DateOfDeathDBpedia/src/main/java/eu/wdaqua/qanary/component/dateofdeathrealpersons/qb/controller/DefaultController.java]: Resolved here: https://github.com/WDAqua/Qanary-question-answering-components/commit/e6d0d608a334a41f53f3bc21fc89b2f85086fe3b

File: qanary-component-QB-DateOfDeathDBpedia/src/main/resources/queries/insert_one_annotation.rq
[the remark this answers is not in the document]
Comment (dschiese) [reply qanary-component-QB-DateOfDeathDBpedia/src/main/resources/queries/insert_one_annotation.rq:1]: Done here, works flawless (thanks for the tip!): https://github.com/WDAqua/Qanary-question-answering-components/pull/303/commits/b972560eadb73d97c4e37a7e81d1e26f6e274691

File: qanary-component-QB-DateOfDeathDBpedia/src/main/java/eu/wdaqua/qanary/component/dateofdeathrealpersons/qb/controller/DefaultController.java (hunk 4 above)
Comment (dschiese) [reply qanary-component-QB-DateOfDeathDBpedia/src/main/java/eu/wdaqua/qanary/component/dateofdeathrealpersons/qb/controller/DefaultController.java]: Done here: https://github.com/WDAqua/Qanary-question-answering-components/commit/e37fbf843c9fb8896f1ccdab6583270d36fd91cb

**Pull request merged**